### ISSUE TITLE
Android `#error` not properly guarded by `if defined(MVPN_ANDROID)`

### ISSUE THREAD

**Issue by Cimbali** [mozilla-mobile/mozilla-vpn-client#4940]
**Describe the bug**
Android-specific checks of Qt version cause builds to fail for non-Android targets, in this case Linux builds. The Android-specific checks should be moved inside the adjacent `#if defined(MVPN_ANDROID)` guards that contain the workarounds.

This is already the case for the same check in  src/commands/commandui.cpp:

https://github.com/mozilla-mobile/mozilla-vpn-client/blob/0f22bdbf08c154c040dbbc4b24f98f0a0ac1928e/src/commands/commandui.cpp#L204-L215

Currently to build the OpenSUSE RPM I have to ship this patch:
```patch
--- mozillavpn-2.11.0~rc8/src/connectionbenchmark/benchmarktaskdownload.cpp	2022-10-31 16:00:04.486905207 +0000
+++ mozillavpn-2.11.0~rc8/src/connectionbenchmark/benchmarktaskdownload.cpp	2022-10-31 16:00:06.858877007 +0000
@@ -47,11 +47,11 @@
     m_dnsLookup.setNameserver(QHostAddress(MULLVAD_DEFAULT_DNS));
 #endif
 
+#if defined(MVPN_ANDROID) || defined(MVPN_WASM)
 #if QT_VERSION >= 0x060500
 #  error Check if QT added support for QDnsLookup::lookup() on Android
 #endif
 
-#if defined(MVPN_ANDROID) || defined(MVPN_WASM)
     NetworkRequest* request =
         NetworkRequest::createForGetUrl(this, m_fileUrl.toString());
     connectNetworkRequest(request);
--- mozillavpn-2.11.0~rc8/src/connectionbenchmark/benchmarktasktransfer.cpp     2022-11-15 16:44:34.537527089 +0000
+++ mozillavpn-2.11.0~rc8/src/connectionbenchmark/benchmarktasktransfer.cpp     2022-11-15 16:44:37.845540228 +0000
@@ -53,7 +53,7 @@
     m_dnsLookup.lookup();
 #endif
 
-#if QT_VERSION >= 0x060400
+#if defined(MVPN_ANDROID) && QT_VERSION >= 0x060400
 #  error Check if QT added support for QDnsLookup::lookup() on Android
 #endif

```

**VPN version:**
To my knowledge, 2.10.1, 2.11.0. maybe earlier versions too.

**Affected Platforms:**
All Linux

**Prerequisites:**
Build the app with newest Qt (currently 6.5, 6.6 when it’s out, etc.)

**Steps to reproduce:**
Just the regular build process, in the extract release directory:
```sh
mkdir -p build 
cd build 
cmake $OLDPWD  -G"Unix Makefiles" -DWEBEXT_INSTALL_LIBDIR:PATH=/usr/lib64 -DCMAKE_INSTALL_SYSCONFDIR:PATH=/etc -DCMAKE_INSTALL_DATADIR:PATH=/usr/share
make
````

**Expected result:**
mozilla VPN builds

**Actual result:**
Build errors out:
```
make[2]: Leaving directory '/builddir/build/BUILD/mozillavpn-2.11.0~rc8/build'
/builddir/build/BUILD/mozillavpn-2.11.0~rc8/src/connectionbenchmark/benchmarktasktransfer.cpp:57:4: error: #error Check if QT added support for QDnsLookup::lookup() on Android
   57 | #  error Check if QT added support for QDnsLookup::lookup() on Android
      |    ^~~~~
make[2]: *** [src/CMakeFiles/mozillavpn.dir/build.make:1093: src/CMakeFiles/mozillavpn.dir/connectionbenchmark/benchmarktasktransfer.cpp.o] Error 1
```

**Additional notes:**
I understand these checks are meant to prevent current workaround code to be built when the workarounds are not needed anymore. Logically the checks also shouldn’t apply when the workaround code is not used at all.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/VPN-3257)

**Comment by oskirby** [comment]
This should now be fixed on the `main` branch by b7813f9ff4df82bebdd6ad27223fc759083a6081

**Comment by data-sync-user** [comment]
➤ Betty Fleming commented:

Owen Kirby can we get story points on this please?
